packages/fxa-auth-server/fxa-oauth-server/lib/routes/authorized-clients/list.js

```diff
@@ -0,0 +1,138 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const hex = require('buf').to.hex;
+const Joi = require('joi');
+
+const config = require('../../config');
+const db = require('../../db');
+const validators = require('../../validators');
+const verifyAssertion = require('../../assertion');
+const ScopeSet = require('fxa-shared').oauth.scopes;
+const localizeTimestamp = require('fxa-shared').l10n.localizeTimestamp({
+  supportedLanguages: config.get('i18n.supportedLanguages'),
+  defaultLanguage: config.get('i18n.defaultLanguage')
+});
+
+
+// Helper function to render each returned record in the expected form.
+function serialize(clientIdHex, token, acceptLanguage) {
+  var creationTime = token.createdAt.getTime();
+  var creationTimeFormatted = localizeTimestamp.format(creationTime, acceptLanguage);
+  var lastAccessTime = token.lastUsedAt.getTime();
+  var lastAccessTimeFormatted = localizeTimestamp.format(lastAccessTime, acceptLanguage);
+  return {
+    client_id: clientIdHex,
+    refresh_token_id: token.refreshTokenId ? hex(token.refreshTokenId) : undefined,
+    client_name: token.clientName,
+    creation_time: creationTime,
+    creation_time_formatted: creationTimeFormatted,
+    last_access_time: lastAccessTime,
+    last_access_time_formatted: lastAccessTimeFormatted,
+    // Sort the scopes alphabetically, for consistent output.
+    scope: token.scope.getScopeValues().sort(),
+  };
+}
+
+
+module.exports = {
+  validate: {
+    payload: {
+      assertion: validators.assertion.required(),
+    }
+  },
+  response: {
+    schema: Joi.array().items(Joi.object({
+      client_id: validators.clientId,
+      refresh_token_id: validators.token.optional(),
+      client_name: Joi.string().required(),
+      creation_time: Joi.number().min(0).required(),
+      creation_time_formatted: Joi.string().optional().allow(''),
+      last_access_time: Joi.number().min(0).required().allow(null),
+      last_access_time_formatted: Joi.string().optional().allow(''),
```

Comment:
Note to self: the front-end doesn't call this endpoint, so no need to do this formatting; it'll be done by auth-server.